pandas/core/categorical.py

```diff
@@ -443,12 +443,18 @@ def _validate_categories(cls, categories):
             raise ValueError('Categorical categories must be unique')
         return categories
 
-    def _set_categories(self, categories):
+    def _set_categories(self, categories, validate=True):
         """ Sets new categories """
```

Comment:
fast path is purely internal - the point is you already have things setup correctly and simply easy to fall the constructor

your example is not valid